src/node_sqlite.cc

```diff
@@ -506,6 +508,132 @@ void StatementSync::All(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(Array::New(isolate, rows.data(), rows.size()));
 }
 
+struct IterateCaptureContext {
+  int num_cols;
+  StatementSync* stmt;
+};
+
+void StatementSync::IterateReturnCallback(
+    const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  auto isolate = env->isolate();
+
+  Local<External> data = Local<External>::Cast(args.Data());
+  IterateCaptureContext* capture_context =
+      static_cast<IterateCaptureContext*>(data->Value());
+  auto stmt = capture_context->stmt;
+  sqlite3_reset(stmt->statement_);
+
+  LocalVector<Name> keys(isolate, {env->done_string(), env->value_string()});
+  LocalVector<Value> values(isolate,
+                            {Boolean::New(isolate, true), Null(isolate)});
+
+  Local<Object> result =
+      Object::New(isolate, Null(isolate), keys.data(), values.data(), 2);
```

Comment:
```suggestion
      Object::New(isolate, Null(isolate), keys.data(), values.data(), values.size());
```

I would also suggest adding a `DCHECK(keys.size() == values.size())` just to be safe.

Reply:
Done (I used `DCHECK_EQ`)